rust/ql/test/library-tests/variables/main.rs

```diff
@@ -342,13 +342,13 @@ fn nested_function() {
     print_i64(f(2)); // $ read_access=f1
 
     {
-        print_i64(f(3)); // $ SPURIOUS: read_access=f1
+        print_i64(f(3));
         fn f(x: i64) -> i64 { // x_3
             2 * x // $ read_access=x_3
         }
 
         {
-            print_i64(f(4)); // $ SPURIOUS: read_access=f1
+            print_i64(f(4));
```

Comment:
Am I understanding correctly that the Rust scoping rules are slightly different for nested functions vs variables - it looks like nested functions are treated as though they're at the beginning of whichever block they're in whereas variables retain their place, then an access binds to the nearest decl above it.

If so the test changes LGTM.  The spurious result we had on line 351 certainly made no sense any way I look at it.

(I've briefly looked at the QL as well and saw no problems, but this library isn't really my area)

Reply:
> Am I understanding correctly that the Rust scoping rules are slightly different for nested functions vs variables - it looks like nested functions are treated as though they're at the beginning of whichever block they're in whereas variables retain their place, then an access binds to the nearest decl above it.

Correct.

Reply:
Thanks.

I'll approve the PR since nobody else is rushing to give a review.  :)